src/Jupyter/Extensions.cs

```diff
@@ -23,6 +25,53 @@ public static void AddIQSharpKernel(this IServiceCollection services)
         {
             services.AddSingleton<ISymbolResolver, Jupyter.SymbolResolver>();
             services.AddSingleton<IExecutionEngine, Jupyter.IQSharpEngine>();
+            services.AddSingleton<IConfigurationSource, ConfigurationSource>();
+        }
+
+        internal static IConfigurationSource ApplyConfiguration<T>(
+            this IConfigurationSource configurationSource,
+            string keyName, Action<T> action
+        )
+        {
+            if (configurationSource.Configuration.TryGetValue(keyName, out var value))
+            {
+                action(value.ToObject<T>());
+            }
+            return configurationSource;
+        }
+
+        public static QuantumSimulator WithJupyterDisplay(this QuantumSimulator simulator, IChannel channel, IConfigurationSource configurationSource)
+        {
+            simulator.DisableLogToConsole();
+            simulator.OnLog += channel.Stdout;
+
+            simulator.Register(
+                typeof(Diagnostics.DumpMachine<>), typeof(JupyterDumpMachine<>),
+                signature: typeof(ICallable)
+            );
+
+            var op = ((GenericCallable)simulator.GetInstance(typeof(Microsoft.Quantum.Diagnostics.DumpMachine<>)));
```

Comment:
oh, wow. it is indeed ugly as hell :(

Reply:
Yeah, sorry about that. I think we just haven't quite exercised this particular use case for `Register` before. As per your offline suggestion, we may be able to remove this ugliness by adding a new event to IQuantumProcessor instead.